pkg/execution/batch/redis.go

```diff
@@ -33,6 +37,43 @@ type redisBatchManager struct {
 	q redis_state.QueueManager
 }
 
+func (b redisBatchManager) batchKey(ctx context.Context, evt event.Event, fn inngest.Function) (string, error) {
+	if fn.EventBatch.Key == nil {
+		return fn.ID.String(), nil
+	}
+
+	out, _, err := expressions.Evaluate(ctx, *fn.EventBatch.Key, map[string]any{"event": evt.Map()})
+	if err != nil {
+		log.From(ctx).Error().Err(err).
+			Str("expression", *fn.EventBatch.Key).
+			Interface("event", evt.Map()).
+			Msg("error evaluating batch key expression")
+		return fn.ID.String(), fmt.Errorf("invalid expression: %w", err)
+	}
+	if str, ok := out.(string); ok {
+		return str, nil
+	}
+	return fmt.Sprintf("%v", out), nil
```

Comment:
ratelimit hashes the key before returning it. could be a something to copy if it makes sense, and can save you that type assertion above.
https://github.com/inngest/inngest/blob/16f8e28ddcb6eca3b28df3319b45f36a7ba363c6/pkg/execution/ratelimit/ratelimit.go#L27-L51

Reply:
I see you're already base64 encoding it, so it might not matter.